File: parser/models/syllabus.py
from typing import List, Optional


class Syllabus:
    def __init__(self, number: str, title: str, content: List[str] = None):
        self.number = number
        self.title = title
        self.content = content if content is not None else []

    def __repr__(self):
        return (
            f"Syllabus(number={self.number}, title={self.title}, content:\n"
            + ",\n".join(self.content)
            + ")"
        )

File: web/backend/models/Syllabus.js
const mongoose = require("mongoose");

const syllabusSchema = new mongoose.Schema(
    {
        number: {
            type: String,
            required: true,
            unique: true,
        },
        title: {
            type: String,
            required: true,
        },
        content: {
            type: [String],
            default: [],
        },
        subject: {
            type: String,
            default: "biology",
        },
    },
    {
        collection: "syllabus",
    }
);

module.exports = syllabusSchema;
